classes/client_helper.php

```diff
@@ -88,6 +88,10 @@ public static function get_client(
      * @return client
      */
     public static function get_csv_client(): client_interface {
-        return new \enrol_oneroster\local\csv_client();
+        $version = get_config('enrol_oneroster', 'oneroster_version');
+
+        if ($version == ' Version 1.2') return new \enrol_oneroster\local\v1p2\csv_client();
```

Comment:
This has a space here, and == shouldnt be used for string comparisons. Does this actually work?